WalletWasabi.Gui/Tabs/WalletManager/LoadWallets/LoadWalletViewModel.cs

```diff
@@ -161,7 +161,7 @@ public KeyManager LoadKeyManager()
 				// Only check requirepassword here, because the above checks are applicable to loadwallet, too and we are using this function from load wallet.
 				if (IsPasswordRequired)
 				{
-					if (PasswordHelper.TryPassword(keyManager, password, out string compatibilityPasswordUsed))
+					if (PasswordHelper.TryPassword(keyManager, password, out string? compatibilityPasswordUsed))
```

Comment:
You should annotate that method `TryPassword` too in the following way:

```csharp
public static bool TryPassword(KeyManager keyManager, string password, out string? compatibilityPasswordUsed)
```
->
```csharp
public static bool TryPassword(KeyManager keyManager, string password, [NotNullWhen(true)] out string? compatibilityPasswordUsed)
```

Very nice documentation is here: https://devblogs.microsoft.com/dotnet/try-out-nullable-reference-types/#conditional-postconditions-maybenullwhenbool-and-notnullwhenbool

Reply:
I didn't do that because then you would get:

![Capture](https://user-images.githubusercontent.com/52379387/92552314-a8bd9380-f260-11ea-87d8-e890459edc76.PNG)

and then you have to change `GetMasterExtKey` method, but its parameter `compatiblityPassword` should be nullable.

Reply:
Yeah, you are right. Sorry.

Could you actually explain how that `TryPassword` function works? It's not obvious to me given complexity of `GetMasterExtKey`.

Reply:
I need an explanation for that as well.
Those functions were introduced in #2100, so I think @molnard can do a better job at explaining it to us.